app/views/contexts/section_summary_context.py

```diff
@@ -169,70 +169,124 @@ def _list_summary_element(self, summary) -> Mapping:
+                "list_name": rendered_summary["for_list"],
+                "related_answers": related_answers,
+                **list_summary_context,
+            }
 
-    def _add_link(self, summary, list_collector_block):
+        if current_list.primary_person:
+            remove_block_id = list_collector_block["remove_block"]["id"]
+            add_link = self._add_link(summary, list_collector_block)
 
-        if list_collector_block:
-            return url_for(
-                "questionnaire.block",
-                list_name=summary["for_list"],
-                block_id=list_collector_block["add_block"]["id"],
-                return_to="section-summary",
+            primary_person_block = self._schema.get_list_collector_for_list(
+                self.section, for_list=summary["for_list"], primary=True
             )
 
-        driving_question_block = QuestionnaireSchema.get_driving_question_for_list(
-            self.section, summary["for_list"]
-        )
+            (
+                primary_person_edit_block_id,
+                edit_block_id,
+            ) = self._get_add_or_edit_blocks_primary(primary_person_block)
 
-        if driving_question_block:
-            return url_for(
-                "questionnaire.block",
-                block_id=driving_question_block["id"],
+            rendered_summary = self._placeholder_renderer.render(
+                summary, self.current_location.list_item_id
+            )
+
+            list_summary_context = self.list_context(
+                list_collector_block["summary"],
+                for_list=list_collector_block["for_list"],
                 return_to="section-summary",
+                edit_block_id=edit_block_id,
+                remove_block_id=remove_block_id,
+                primary_person_edit_block_id=primary_person_edit_block_id,
+                for_list_item_ids=current_list.primary_person,
             )
 
+            related_answers = self._get_related_answers(current_list.first)
+
+            return {
+                "title": rendered_summary["title"],
+                "type": rendered_summary["type"],
+                "add_link": add_link,
+                "add_link_text": rendered_summary["add_link_text"],
+                "empty_list_text": rendered_summary.get("empty_list_text"),
+                "list_name": rendered_summary["for_list"],
+                "related_answers": related_answers,
+                **list_summary_context,
+            }
+
+    @staticmethod
+    def _add_link(summary, list_collector_block):
+
+        return url_for(
+            "questionnaire.block",
+            list_name=summary["for_list"],
+            block_id=list_collector_block["add_block"]["id"],
+            return_to="section-summary",
+        )
+
     def _get_safe_page_title(self, title):
         return (
             safe_content(self._schema.get_single_string_value(title)) if title else ""
         )
+
+    def _get_related_answers(self, list_item):
```

Comment:
Need to add type hints for the new methods added

Reply:
Added some type hints (newly added methods and some old ones), thanks!